client/src/pages/Projects.jsx

```diff
@@ -22,17 +39,219 @@ const Projects = (): Node => {
+          teamEmail = { ...teamEmail, [e.email]: e.firstName + e.lastName };
+          teamMemberList.push({
+            key: e.firstName + e.lastName,
+            value: e.firstName + e.lastName,
+            text: `${e.firstName} ${e.lastName}`,
+          });
+        });
+        setTeamMembers(teamMemberList);
+        setTeamEmails(teamEmail);
+      }
+    };
+    const getCurrMember = async () => {
+      const member = await getUserAuth();
+      setEditable(
+        member.data &&
+          (member.data.result.level === 'ADMIN' ||
+            member.data.result.level === 'DIRECTOR'),
+      );
+    };
     getAllProjects();
+    getAllMembers();
+    getCurrMember();
   }, []);
 
+  const filterObj = (raw, allowed) => {
+    const filtered = Object.values(
+      Object.fromEntries(
+        Object.entries(raw).filter(([key]) => allowed.includes(key)),
+      ),
+    );
+    return filtered;
+  };
+
   return (
-    <Page title="Projects">
-      <Table
-        data={projects}
-        columns={projectColumnDefs}
-        onRowClick={(e) => history.push(`/projects/${e.data._id}`)}
+    <Sidebar.Pushable as={Segment} style={{ height: '100vh' }}>
+      <Modal
+        open={showModal}
+        header="You have unsaved changes"
+        size="mini"
+        content="Save your changes before going back?"
+        actions={[
+          {
+            key: 'no',
+            content: 'No',
+            negative: true,
+            onClick: () => setShowModal(false),
+          },
+          {
+            key: 'Save',
+            content: 'Save',
+            positive: true,
+            icon: 'check',
+            labelPosition: 'right',
+            onClick: () => {
+              setUnmodProj(currProj);
+              updateProject(currProj, currProj._id);
+              setShowModal(false);
+              window.location.reload();
+            },
+          },
+        ]}
       />
-    </Page>
+      <Sidebar
+        as="form"
+        animation="overlay"
+        onHide={() => {
+          setVisible(false);
+          setEditMode(false);
+          if (!isEqual(currProj, unmodProj)) setShowModal(true);
+        }}
+        visible={visible}
+        width="wide"
+        direction="right"
+      >
+        <Icon
+          name="arrow left"
+          onClick={() => {
+            setVisible(false);
+            setEditMode(false);
+          }}
+          style={{ cursor: 'pointer' }}
+        />
+        {editable && (
+          <Icon
+            name="edit outline"
+            style={{ cursor: 'pointer' }}
+            onClick={() => setEditMode(true)}
+          />
+        )}
+        <Form.Group>
+          <Form.Input
+            label="Project Name"
+            value={currProj.projectName}
+            onChange={(e, { value }) => {
+              setCurrProj({ ...currProj, projectName: value });
+            }}
+            readOnly={!editMode}
+          />
+          <Form.Dropdown
```

Comment:
nit: Theres a lot of `<Form.Group>, <Form.Input>, <Form.Dropdown>` ... etc so destructuring might make it cleaner just fyi
`const { Group, Input, Dropdown, ...etc } = Form`
and then just use it in the return like so:
`<Group>, <Input> <Dropdown>`